src/classes/metadata/pdf.vala

```diff
@@ -363,17 +363,35 @@ namespace pdfpc.Metadata {
          * Fill the slide notes from pdf text annotations.
          */
         private void notes_from_document() {
-            for(int i = 0; i < this.page_count; i++) {
+            for (int i = 0; i < this.page_count; i++) {
                 var page = this.document.get_page(i);
+                int user_slide = real_slide_to_user_slide(i);
+                string note_text = this.notes.get_note_for_slide(user_slide);
```

Comment:
I would make it clear, that `note_text` comes from the pdfpc file. Like
`this.notes.get_note_from_pdfpc_for_slide` and `this.notes.get_note_from_pdf_for_slide`

Reply:
Actually, this.notes contain both types of notes, and so get_note_for_slide may return either. There is a is_note_native() method to tell them apart. I was hesitating whether to alter get_note_for_slide() to return the entire slide_note structure instead...